docs/user_guide/ml.md

```diff
@@ -1,8 +1,254 @@
+This flag determines if a certain `dataset` and shift combination can be used by the task.
+
+```{literalinclude} ./ml_code.py
+:language: python
+:start-at: def evaluate
+:end-at: return events
+```
+
+The evaluations output is saved as parquet file with the name of the model as field name.
+To get the files path afterwards, rerun the same law command again, but with `--print-output 0` at the end.
+
+# Commands to start training and evaluation
+To start the machine learning training `law run cf.MLTraining`, while for evaluation use `law run cf.MLEvaluation`.
+When not using the `config`, `calibrators`, `selector`s or `dataset`
+```bash
+law run cf.MLTraining \
+    --version your_current_version \
+    --ml-model test_model_name \
+    --config run2_2017_nano_uhh_v11_limited \
+    --calibrators calibrator_name \
+    --selector selector_name \
+    --dataset datasets_1,dataset_2,dataset_3,...
+
+law run cf.MLEvaluation \
+    --version v1 \
+    --ml-model test_model \
+    --config run2_2017_nano_uhh_v11_limited \
+    --calibrators skip_jecunc
+```
+Most of these settings should sound familiar, if not look into the corresponding tutorial.
+`version` defines a setup configuration of your ML task, think more of a label than of an actual `version`.
+If you change the label, you
+
+# Optional useful functions:
+## Separate training and evaluation configuration for configs, calibrators, selector and producers:
+By default chosen configs, calibrators, selector and producer are used for both training and evaluation.
+Sometimes one do not want to share the same environment or does not need all the columns in evaluation as in training, this is where a separation of both comes in handy.
+
+To separate this behavior one need to define the training_{configs,calibrators, selector,producers}.
+These function take always the `config_inst` as first, and the requested_{configs,calibrators, selector,producers} as second parameter.
+If this function is defined the evaluation will use the give `config`,`calibrator`, `selector` or `producer`, while the training will use one defined in the function.
+
+In the following case, training will use a fixed selector and producer called default, while the calibrator uses an own defined calibrator called "skip_jecunc":
+
+```{literalinclude} ./ml_code.py
+:language: python
+:start-at: def training_selector
+:end-at: return ["default"]
+```
+```{literalinclude} ./ml_code.py
+:language: python
+:start-at: def training_producers
+:end-at: return ["default"]
+```
+```{literalinclude} ./ml_code.py
+:language: python
+:start-at: def training_calibrators
+:end-at: return ["skip_jecunc"]
+```
+## setup
+Setup is called at the end of `__init__` of your ML model.
+Within this function you can prepare operations that should stay open during the whole life time of your instance.
+An typical example for this would the opening of a file or dynamic appending of variables that are only relevant to your ML model, and no where else.
```

Comment:
```suggestion
A typical example for this would be the opening of a file or dynamically appending variables that are only relevant to your ML model, and no where else.
```